distributed/tests/test_cancelled_state.py

```diff
@@ -198,6 +194,9 @@ async def wait_and_raise(*args, **kwargs):
 
     fut = c.submit(wait_and_raise)
     await wait_for_state(fut.key, "executing", w)
+    # Queue up another task to ensure this is not affected by our error handling
+    fut2 = c.submit(inc, 1)
+    await wait_for_state(fut2.key, "ready", w)
```

Comment:
@crusaderky this triggers the condition you are concerned about in the transition function about dropped instructions.

The task is queued up and we'll receive a recommendation. The only instruction at that point is the TaskErred message.